### PR TITLE
Adjust to membrane_core v0.12

### PR DESCRIPTION
**Pull request by FelonEkonom**

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -38,8 +38,8 @@ defmodule Membrane.ICE.Mixfile do
 
   defp deps do
     [
+      {:membrane_core, "~> 0.12.0", override: true},
```
Comment (mickel8) [inline mix.exs:41]: Remove override before merging

File: lib/membrane_ice_plugin/ice_endpoint.ex [hunk 2]
```diff
@@ -168,7 +168,7 @@ defmodule Membrane.ICE.Endpoint do
     availability: :on_request,
     accepted_format: _any,
     mode: :pull,
-    demand_unit: :buffers
+    demand_mode: :auto
```
Comment (mickel8) [inline lib/membrane_ice_plugin/ice_endpoint.ex:171]: I assume that the old `mode` field is redundant? 

Also, in the previous version we were guaranteed not to receive demands until ICE connection was established. Do we know how `auto` mode will behave?

**Review by mickel8 — COMMENTED** [review]
There are a couple of references to `demand` term in the code so we migth want to delete them too

File: lib/membrane_ice_plugin/ice_endpoint.ex (hunk 2 above)
Comment (FelonEkonom) [reply lib/membrane_ice_plugin/ice_endpoint.ex:171]: After establishing an ICE connection `ICE.Endpoint` sends keying material in upstream `event` to the encryptor. Encryptor won't send buffers to ICE before keying material arrival

**Pull request closed without merging**